webtau-http/src/test/java/com/twosigma/webtau/http/HttpJavaTest.java

```diff
@@ -0,0 +1,181 @@
+import java.util.regex.Pattern;
+
+import static com.twosigma.webtau.Ddjt.*;
+import static com.twosigma.webtau.http.Http.http;
+import static com.twosigma.webtau.utils.CollectionUtils.createMap;
+
+public class HttpJavaTest implements HttpConfiguration  {
+    private static final HttpTestDataServer testServer = new HttpTestDataServer();
+
+    private static final int PORT = 7824;
+
+    @BeforeClass
+    public static void startServer() {
+        testServer.start(PORT);
+    }
+
+    @AfterClass
+    public static void stopServer() {
+        testServer.stop();
+    }
+
+    @Before
+    public void initCfg() {
+        HttpConfigurations.add(this);
+    }
+
+    @After
+    public void cleanCfg() {
+        HttpConfigurations.remove(this);
+    }
+
+    @Test
+    public void simpleObjectMappingExample() {
+        http.get("/end-point-simple-object", (header, body) -> {
+            body.get("k1").should(equal("v1"));
+        });
+    }
+
+    @Test
+    public void simpleListMappingExample() {
+        http.get("/end-point-simple-list", (header, body) -> {
+            body.get(0).get("k1").should(equal("v1"));
+        });
+    }
+
+    @Test
+    public void equalityMatcher() {
+        http.get("/end-point", (header, body) -> {
+            body.get("id").shouldNot(equal(0));
+            body.get("amount").should(equal(30));
+
+            body.get("list").should(equal(Arrays.asList(1, 2, 3)));
+
+            body.get("object").get("k1").should(equal(
+                    Pattern.compile("v\\d"))); // regular expression matching
+
+            body.get("object").should(equal(createMap("k1", "v1", "k3", "v3"))); // matching only specified fields and can be nested multiple times
+
+            body.get("complexList").should(equal(header("k1" , "k2").values( // matching only specified fields, but number of entries must be exact
+                                                        "v1" ,  30,
+                                                        "v11",  40)));
+        });
+
+        http.doc.capture("end-point-object-equality-matchers");
+    }
+
+    @Test
+    public void compareNumbersWithGreaterLessMatchers() {
+        http.get("/end-point-numbers", (header, body) -> {
+            body.get("id").shouldBe(greaterThan(0));
+            body.get("price").shouldBe(greaterThanOrEqual(100));
+            body.get("amount").shouldBe(lessThan(150));
+            body.get("list").get(1).shouldBe(lessThanOrEqual(2));
+
+            body.get("id").shouldNotBe(lessThanOrEqual(0));
+            body.get("price").shouldNotBe(lessThan(100));
+            body.get("amount").shouldNotBe(greaterThanOrEqual(150));
+            body.get("list").get(1).shouldNotBe(greaterThan(2));
+        });
+
+        http.doc.capture("end-point-numbers-matchers");
+    }
+
+    @Test
+    public void containMatcher() {
+        http.get("/end-point-list", (header, body) -> {
+            body.should(contain(createMap("k1", "v1", "k2", "v2")));
```

Comment:
Maybe this would read better if `createMap` was renamed to just `map`?

Reply:
same thought occurred. `map` is a functional paradigm as well and may be confusing. Give me more ideas please.

Reply:
`aMapOf()`?